src/history/HistoryArchive.cpp

```diff
@@ -275,17 +276,35 @@ HistoryArchiveState::HistoryArchiveState() : server(STELLAR_CORE_VERSION)
 }
 
 HistoryArchiveState::HistoryArchiveState(uint32_t ledgerSeq,
-                                         BucketList const& buckets)
+                                         BucketList const& buckets,
+                                         bool clearFutureBuckets)
```

Comment:
I don't like this `clearFutureBuckets` thingy: why do we set shadows in the first place?